docs/src/use/configure/configuration-files.md

```diff
@@ -453,3 +456,15 @@ npx eslint --config some-other-file.js **/*.js
 ```
 
 In this case, ESLint does not search for `eslint.config.js` and instead uses `some-other-file.js`.
+
+## TypeScript Support
+
+ESLint supports loading configuration files written in TypeScript optionally. The TypeScript support is powered by [`importx`](https://github.com/antfu/importx), which does not come out-of-the-box with ESLint and you will need to install it as a dev dependency first:
+
+```shell
+npm install --save-dev importx
+```
+
+Note that `importx` only strips off type annotations and does not perform type checking.
+
+When both `eslint.config.js` and `eslint.config.ts` are present in the same directory, JavaScript versions always take precedence unless provided specifically via `--config` option.
```

Comment:
We should note in the docs that this feature is still experimental, so it could change anytime.